src/workflows/airqo_etl_utils/datautils.py

```
@@ -732,18 +732,26 @@ def _extract_device_api_data(
+                        mapping = config["mapping"][network]
+                        result = data_source_api.air_gradient(device, dates)
+
+                if result.data:
+                    return DataUtils.map_and_extract_data(mapping, result.data), {}
+                else:
+                    logger.info(
+                        f"No data returned from {device.get('device_id')} for the given date range"
+                    )
             except Exception as e:
                 logger.exception(
                     f"An error occurred: {e} - device {device.get('name')}"
```

Ruff (0.14.3)

Redundant exception object included in `logging.exception` call

(TRY401)